src/model/kubernetes.js

```diff
@@ -0,0 +1,329 @@
+                    value: this.buildParameters.buildMethod,
+                  },
+                  {
+                    name: 'CUSTOM_PARAMETERS',
+                    value: this.buildParameters.customParameters,
+                  },
+                  {
+                    name: 'BUILD_TARGET',
+                    value: this.buildParameters.platform,
+                  },
+                  {
+                    name: 'ANDROID_VERSION_CODE',
+                    value: this.buildParameters.androidVersionCode.toString(),
+                  },
+                  {
+                    name: 'ANDROID_KEYSTORE_NAME',
+                    value: this.buildParameters.androidKeystoreName,
+                  },
+                  {
+                    name: 'ANDROID_KEYALIAS_NAME',
+                    value: this.buildParameters.androidKeyaliasName,
+                  },
+                ],
+                volumeMounts: [
+                  {
+                    name: 'data',
+                    mountPath: '/data',
+                  },
+                  {
+                    name: 'credentials',
+                    mountPath: '/credentials',
+                    readOnly: true,
+                  },
+                ],
+                lifeCycle: {
+                  preStop: {
+                    exec: {
+                      command: [
+                        'bin/bash',
+                        '-c',
+                        `cd /data/builder/action/steps;
+                        chmod +x /return_license.sh;
+                        /return_license.sh;`,
+                      ],
+                    },
+                  },
+                },
+              },
+            ],
+            restartPolicy: 'Never',
+          },
+        },
+        backoffLimit: 1,
+      },
+    };
+    await this.kubeClient.apis.batch.v1.namespaces('default').jobs.post({ body: jobManifest });
+    core.info('Job created');
+  }
+
+  static async watchBuildJobUntilFinished() {
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+
+    let podname;
+    let ready = false;
+    while (!ready) {
+      // eslint-disable-next-line no-await-in-loop
+      const pods = await this.kubeClient.api.v1.namespaces('default').pods.get();
+      // eslint-disable-next-line no-plusplus
+      for (let index = 0; index < pods.body.items.length; index++) {
+        const element = pods.body.items[index];
+        if (element.metadata.labels['job-name'] === this.jobName) {
+          if (element.status.phase !== 'Pending') {
+            core.info('Pod no longer pending');
+            if (element.status.phase === 'Failure') {
+              core.error('Kubernetes job failed');
+            } else {
+              ready = true;
+              podname = element.metadata.name;
+            }
+          }
+        }
+      }
+    }
+
+    if (!podname) {
+      core.error('no pods to stream logs from found');
```

Comment:
Could you please streamline this sentence?

Reply:
This check wasn't needed, cleaned up.